bucket.go

```diff
@@ -33,6 +33,7 @@ type Bucket interface {
 	Set(k string, exp int, v interface{}) error
 	SetRaw(k string, exp int, v []byte) error
 	Delete(k string) error
+	Remove(k string, cas uint64) (casOut uint64, err error)
```

Comment:
Actually can you add a comment that describes how a `Remove()` is distinct from a `Delete()`.  At face value, they sound like they do exactly the same thing?